### PR TITLE
quattor/functions/network: cleanup and removal of get_subnet_params function

### LINKED ISSUE #148: quattor/functions/network: boot_nic returns last, not first boot device

### PR DESCRIPTION
**Pull request by stdweird**
Fixes #148 #149
Backwards incompatible due to removal of get_subnet_params function #

### CODE REVIEW

File: quattor/functions/network.pan [hunk 1]
```diff
@@ -1,206 +1,123 @@
...
+}
+function copy_network_params = {
+    nics = value('/hardware/cards/nic');
 
-    if ( !is_list(subnet_list) ) {
-        error("get_subnet_params first argument must be a list");
+    if (!is_defined(nics)) {
+        error("No network cards defined in the configuration");
+    };
+    if (!is_nlist(nics)) {
```
Comment (ned21) [inline quattor/functions/network.pan]: It's confusing to refer to this as a dict everywhere and then use is_nlist() in preference to is_dict.

**Comment by stdweird** [comment]
@ned21 ah, and there's even an `is_dict` function (didn't know that, had to look it up 😞 )
@jrha can you add mandatory `is_dict` instead of `is_nlist` to panlint?

**Comment by stdweird** [comment]
@ned21 fixed

**Comment by jouvin** [comment]
I'm late to review this merged PR... I am a bit surprised that it has been merged as it. In particular, I think that as a general rule we should not merge a PR flagged as backward incompatible without a few word in its description summarizing what is the backward incompatibility (done in this PR) and the motivation for this change (missing). In this case, the removal of `get_subnet_params()` is a problem for GRIF configuration... I'll look in more details and open an issue if needed.

**Comment by stdweird** [comment]
it wasn't really clear to me what the code did (it looks complicated for something trivial)
It also had
This is a 'dirty solution' to work around the lack of bitwise AND in pan...
in the comment which is not true. 
I guess we can readd the function if needed, but code should be cleaned up at least.

**Comment by jouvin** [comment]
I think this is really old stuff that I wrote... to make easier to have machines with different network configurations (in terms of subnets, routers...) and retrieve the information based on the machine IP, without adding any specific config. The comment about the lack of bitwise AND in panc was true at that time I remember!
I'll look at reimplement the function in a modernized way...

**Comment by jrha** [comment]
@jouvin see #151.

**Pull request merged**